planex/fileupdate.py

```diff
@@ -0,0 +1,62 @@
+"""
+<pipe> | idem <file> overwrites <file> with the output of <pipe>
+if the output of <pipe> is different from the contents of <file>,
+otherwise leaves <file> untouched.
+"""
+
+import hashlib
+import os
+import shutil
+import sys
+import tempfile
+
+def sameSize(infile, outfile):
+    infile.seek(0)
```

Comment:
We need to sync(), otherwise os.fstat() sees a zero-length file:
```
$ python
Python 2.7.5 (default, Nov  6 2016, 00:28:07) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> infile = open("foo", "wb")
>>> infile.writelines("ntahusnthsnthstnh\n")
>>> import os
>>> os.fstat(infile.fileno()).st_size
0
```

Alternatively we can use tell(), but we will first need to seek() to the end of the file because we  the caller may have called seek() after writing to the file descriptor.